tools/stripdebug.ml

```diff
@@ -23,27 +23,24 @@ open Misc
 
 let stripdebug infile outfile =
   let ic = open_in_bin infile in
-  Bytesections.read_toc ic;
-  let toc = Bytesections.toc() in
-  let pos_first_section = Bytesections.pos_first_section ic in
+  let toc = Bytesections.read_toc ic in
   let oc =
     open_out_gen [Open_wronly; Open_creat; Open_trunc; Open_binary] 0o777
                  outfile in
   (* Skip the #! header, going straight to the first section. *)
-  seek_in ic pos_first_section;
   (* Copy each section except DBUG and CRCS *)
-  Bytesections.init_record oc;
+  let toc_writer = Bytesections.init_record oc in
   List.iter
-    (fun (name, len) ->
-      if name = "DBUG" || name = "CRCS" then begin
-        seek_in ic (in_channel_length ic + len)
```

Comment:
This looks like a bug.

Reply:
It is, yes - I'd hit while working on the relocatable patches. In practice it doesn't cause problem because of the order in which the sections are encountered. I'd just switched to `pos_in`, which it looks like is what was intended.

Reply:
The new implementation always seek to the beginning of the section before copying the data.

Reply:
This was fixed by @xavierleroy in f2c2c15dd1123436073398c002e74bf2d1044109